flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/databases/snowflake/dialect/SnowflakeRowConverter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.jdbc.databases.snowflake.dialect;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.connector.jdbc.converter.AbstractPostgresCompatibleRowConverter;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.postgresql.jdbc.PgArray;
+
+/**
+ * Runtime converter that responsible to convert between JDBC object and Flink internal object for
+ * Snowflake.
+ */
+@Internal
+public class SnowflakeRowConverter extends AbstractPostgresCompatibleRowConverter<PgArray> {
+
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public String converterName() {
+        return "Snowflake";
+    }
+
+    // https://docs.snowflake.com/en/sql-reference/intro-summary-data-types
+    public SnowflakeRowConverter(RowType rowType) {
```

Comment:
I am not sure exactly how Snowflake SQL compares to the other SQLS. It seems to be close to SQL server , which has some TINYINT logic in its converter - it looks like Snowflake supports this.

I also see [https://docs.uipath.com/process-mining/automation-suite/2022.10/user-guide/sql-differences-between-snowflake-and-sql-server](https://docs.uipath.com/process-mining/automation-suite/2022.10/user-guide/sql-differences-between-snowflake-and-sql-server). Do any of these SQL quirks require converter changes? 

I notice that Snowflake SQL uses double quotes, where as Flink SQL uses back ticks , it would be worth testing that quotes work for Snowflake.

Reply:
> I am not sure exactly how Snowflake SQL compares to the other SQLS. It seems to be close to SQL server , which has some TINYINT logic in its converter - it looks like Snowflake supports this type - so you need to add this to the converter.
> 
> I also see https://docs.uipath.com/process-mining/automation-suite/2022.10/user-guide/sql-differences-between-snowflake-and-sql-server. Do any of these SQL quirks require converter changes?
> 
> I notice that Snowflake SQL uses double quotes, where as Flink SQL uses back ticks , it would be worth testing that quotes work for Snowflake.

https://docs.snowflake.com/en/sql-reference/intro-summary-data-types

This code was tested with Flink, so whatever Flink does should work. I've provided an example in the PR.

Reply:
@davidradl Could you please take a look at this PR?